integrations/opensearch/src/haystack_integrations/components/retrievers/opensearch/open_search_hybrid_retriever.py

```diff
@@ -0,0 +1,286 @@
+# SPDX-FileCopyrightText: 2023-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+from typing import Any, Dict, List, Optional, Union
+
+from haystack import DeserializationError, Pipeline, default_from_dict, default_to_dict, logging, super_component
+from haystack.components.embedders.types import TextEmbedder
+from haystack.components.joiners import DocumentJoiner
+from haystack.components.joiners.document_joiner import JoinMode
+from haystack.core.serialization import component_from_dict, import_class_by_name
+from haystack.document_stores.types import FilterPolicy
+
+from haystack_integrations.components.retrievers.opensearch import OpenSearchBM25Retriever, OpenSearchEmbeddingRetriever
+from haystack_integrations.document_stores.opensearch import OpenSearchDocumentStore
+
+logger = logging.getLogger(__name__)
+
+
+@super_component
+class OpenSearchHybridRetriever:
+    """
+    A hybrid retriever that combines embedding-based and keyword-based retrieval from OpenSearch.
+
+    This component requires haystack-ai>=2.13.0 to work properly.
+    """
```

Comment:
Let's add a usage example here. Can be the same as in the integrations page, once we have it.